bids/analysis/analysis.py

```diff
@@ -203,17 +203,29 @@ def setup(self, input_nodes=None, auto_contrasts=None, **kwargs):
 
             model = self.model or {}
 
-            variables = set(model.get('variables', []))
             hrf_variables = set(model.get('HRF_variables', []))
-            if not variables >= hrf_variables:
+            if 'explanatory_variables' in model:
+                evs = model['explanatory_variables']
+                nuisance = model.get('nuisance_variables', [])
```

Comment:
This is a legacy-vs-new check, so it will go away and we can use the same treatment.

That said, I'm specifically proposing that to estimate or contrast a variable makes that variable "of interest", even if it is a confound like `FramewiseDisplacement`. The only situation I can think of that would not produce contrasts is something like regfilt, where we expect to produce a "cleaned" dataset. I'm not sure how that would fit into BIDS model/FitLins.

Reply:
Every column included in the design matrix is estimated, so there's no meaningful distinction between explanatory and nuisance variables at the model level. The distinction only makes sense if we use it as shorthand for indicating which variables the `auto_contrasts` flag applies to.

I guess I still find myself going back and forth on this. It seems more principled to have just a single `variables` field that includes all named variables to be included, and then a separate `HRF_variables` field that indicates the subset to be convolved. What seems most explicit and consistent with that pattern to me would be to make `auto_contrasts` a list of variable names rather than a boolean. The downside is redundancy: if writing models by hand, the user may end up writing most of the variable names three times (once in `variables`, once in `HRF_variables`, and once in `auto_contrasts`). But if we think that users aren't going to be writing models manually in the long term, then maybe this is the right way to go. From a conceptual standpoint, I think it's the simplest model of what's going on, and doesn't force any distinctions on the user that they may not respect (e.g., explanatory vs. nuisance). Anyway, yeah, let's discuss this at Stanford.